tiledb/sm/query/query.cc

```diff
@@ -1063,11 +1168,82 @@ Status Query::set_subarray_unsafe(const NDRange& subarray) {
   return Status::Ok();
 }
 
+Status Query::check_buffers_validity() {
+  // Iterate through each attribute
+  for (auto& attr : buffer_names()) {
```

Comment:
for all of these checks, let's return an error message that explicitly tells the user which buffer is unset, or over-set. i think the logic should be:
```
for (auto& attr : buffer_names()) {
  // 1. error if data buffer not set
  // 2a. if var-sized, error if offsets buffer not set
  // 2b. if fixed-size, error if offsets buffer is set
  // 3a. if nullable, error if validity buffer not set
  // 3b. if non-nullable, error if validity buffer is set.
}
```